pkg/apis/apps/kfdef/v1beta1/application_types.go

```diff
@@ -0,0 +1,170 @@
+type Application struct {
+	Name            string           `json:"name,omitempty"`
+	KustomizeConfig *KustomizeConfig `json:"kustomizeConfig,omitempty"`
+}
+
+type KustomizeConfig struct {
+	RepoRef    *RepoRef    `json:"repoRef,omitempty"`
+	Overlays   []string    `json:"overlays,omitempty"`
+	Parameters []NameValue `json:"parameters,omitempty"`
+}
+
+type RepoRef struct {
+	Name string `json:"name,omitempty"`
+	Path string `json:"path,omitempty"`
+}
+
+type NameValue struct {
+	Name         string `json:"name,omitempty"`
+	Value        string `json:"value,omitempty"`
+	InitRequired bool   `json:"initRequired,omitempty"`
+}
+
+// Plugin can be used to customize the generation and deployment of Kubeflow
+type Plugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec *runtime.Object `json:"spec,omitempty"`
+}
+
+// Secret provides information about secrets needed to configure Kubeflow.
+// Secrets can be provided via references e.g. a URI so that they won't
+// be serialized as part of the KfDefSpec which is intended to be written into source control.
+type Secret struct {
+	Name         string        `json:"name,omitempty"`
+	SecretSource *SecretSource `json:"secretSource,omitempty"`
+}
+
+type SecretSource struct {
+	LiteralSource *LiteralSource `json:"literalSource,omitempty"`
+	HashedSource  *HashedSource  `json:"hashedSource,omitempty"`
+	EnvSource     *EnvSource     `json:"envSource,omitempty"`
+}
+
+type LiteralSource struct {
+	Value string `json:"value,omitempty"`
+}
+
+type HashedSource struct {
+	HashedValue string `json:"value,omitempty"`
+}
+
+type EnvSource struct {
+	Name string `json:"Name,omitempty"`
+}
+
+// Repo provides information about a repository providing config (e.g. kustomize packages,
+// Deployment manager configs, etc...)
+type Repo struct {
+	// Name is a name to identify the repository.
+	Name string `json:"name,omitempty"`
+	// URI where repository can be obtained.
+	// Can use any URI understood by go-getter:
+	// https://github.com/hashicorp/go-getter/blob/master/README.md#installation-and-usage
+	Uri string `json:"uri,omitempty"`
+}
+
+// KfDefStatus defines the observed state of KfDef
+type KfDefStatus struct {
+	Conditions []KfDefCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,6,rep,name=conditions"`
+	// ReposCache is used to cache information about local caching of the URIs.
+	ReposCache map[string]RepoCache `json:"reposCache,omitempty"`
+}
+
+type RepoCache struct {
+	LocalPath string `json:"localPath,string"`
+}
+
+type KfDefConditionType string
+
+const (
+	// KfAvailable means Kubeflow is serving.
+	KfAvailable KfDefConditionType = "Available"
+
+	// KfDegraded means functionality of Kubeflow is limited.
+	KfDegraded KfDefConditionType = "Degraded"
+
+	// KfPluginsProgressing means kfctl is applying plugins.
+	KfPluginsProgressing KfDefConditionType = "PluginsProgressing"
```

Comment:
Do we need PluginsProcessing? Should we hold off defining it until we know we need it?

Reply:
removed.